### DISCUSSION TITLE
Error in fitting model

### DISCUSSION THREAD

**Discussion by eeinnew** [teamalgoritma/community#140] [Q&A] [answered]
Hello, I am about to fit my model with this code below.
```
history_base <-   model_base %>% 
  fit(x = train_x,
      y = train_y, 
      epoch = 10, 
      validation_data = list(test_x, test_y), 
      verbose = 1,
      batch_size = 150,
      shuffle  = F)
```

however it is error 
```
Error in py_call_impl(callable, dots$args, dots$keywords) : 
ValueError: in user code:
<... omitted ...>R-TENS~1\lib\site-packages\keras\engine\compile_utils.py", line 201, in __call__
loss_value = loss_obj(y_t, y_p, sample_weight=sw)
File "C:\Users\LENOVO\ANACON~1\envs\R-TENS~1\lib\site-packages\keras\losses.py", line 139, in __call__
losses = call_fn(y_true, y_pred)
File "C:\Users\LENOVO\ANACON~1\envs\R-TENS~1\lib\site-packages\keras\losses.py", line 243, in call **
return ag_fn(y_true, y_pred, **self._fn_kwargs)
File "C:\Users\LENOVO\ANACON~1\envs\R-TENS~1\lib\site-packages\keras\losses.py", line 1788, in categorical_crossentropy
y_true, y_pred, from_logits=from_logits, axis=axis)
File "C:\Users\LENOVO\ANACON~1\envs\R-TENS~1\lib\site-packages\keras\backend.py", line 5119, in categorical_crossentropy
target.shape.assert_is_compatible_with(output.shape)

ValueError: Shapes (None, 10, 24) and (None, 24) are incompatible

See `reticulate::py_last_error()` for details
```

**Comment by hwulanayu** [comment, marked as answer]
Dear @eeinnew,

error tersebut terjadi karena kesalahan memberikan bentuk/shape array yang salah pada model Anda. Keras mengharapkan array shape (None, 10, 24), sementara Anda memberikan array bentuk (None, 24). Pastikan:

1. pemilihan loss function sesuai dengan jumlah label target `binary_crossentropy` untuk 2 kelas, `categorical_crossentropy` untuk multi kelas > 2)
2. jumlah node pada **output layer** sesuai dengan jumlah target kelas dan tidak ada parameter `input_shape`.
3. label target (y) sudah dalam bentuk one-hot-encoding dengan fungsi `to_categorical()`

Jika hal di atas sudah dicek dan tidak ada yang salah, silakan untuk mencantumkan `summary()` dari arsitektur model dan dimensi train-test-split datanya.
